eodag/rest/errors.py

```diff
@@ -0,0 +1,172 @@
+# limitations under the License.
+import logging
+from typing import Dict, List, Tuple, Union
+
+from fastapi import FastAPI, Request
+from fastapi.responses import ORJSONResponse
+from starlette import status
+from starlette.exceptions import HTTPException as StarletteHTTPException
+
+from eodag.rest.types.eodag_search import EODAGSearch
+from eodag.utils.exceptions import (
+    AuthenticationError,
+    DownloadError,
+    EodagError,
+    MisconfiguredError,
+    NoMatchingProductType,
+    NotAvailableError,
+    RequestError,
+    TimeOutError,
+    UnsupportedProductType,
+    UnsupportedProvider,
+    ValidationError,
+)
+
+EODAG_DEFAULT_STATUS_CODES = {
+    AuthenticationError: status.HTTP_500_INTERNAL_SERVER_ERROR,
+    DownloadError: status.HTTP_500_INTERNAL_SERVER_ERROR,
+    MisconfiguredError: status.HTTP_500_INTERNAL_SERVER_ERROR,
+    NotAvailableError: status.HTTP_404_NOT_FOUND,
+    NoMatchingProductType: status.HTTP_404_NOT_FOUND,
+    TimeOutError: status.HTTP_504_GATEWAY_TIMEOUT,
+    UnsupportedProductType: status.HTTP_404_NOT_FOUND,
+    UnsupportedProvider: status.HTTP_404_NOT_FOUND,
+    ValidationError: status.HTTP_400_BAD_REQUEST,
+}
+
+logger = logging.getLogger("eodag.rest.server")
+
+
+class ResponseSearchError(Exception):
+    """Represent a EODAG search error response"""
+
+    def __init__(self, errors: List[Tuple[str, Exception]]) -> None:
+        self._errors = errors
+
+    @property
+    def errors(self) -> List[Dict[str, Union[str, int]]]:
+        """return errors as a list of dict"""
+        error_list: List[Dict[str, Union[str, int]]] = []
+        for name, exception in self._errors:
+
+            error_dict = {"provider": name, "error": exception.__class__.__name__}
+
+            if exception.args:
+                error_dict["message"] = exception.args[0]
+
+            if len(exception.args) > 1:
+                error_dict["detail"] = " ".join(exception.args[1:])
+
+            error_dict["status_code"] = (
+                EODAG_DEFAULT_STATUS_CODES.get(
+                    type(exception), getattr(exception, "status_code", None)
+                )
+                or 500
+            )
+
+            if type(exception) in (MisconfiguredError, AuthenticationError):
+                logger.error("%s: %s", type(exception).__name__, str(exception))
+                error_dict[
+                    "message"
+                ] = "Internal server error: please contact the administrator"
+                error_dict.pop("detail", None)
+
+            if params := getattr(exception, "parameters", None):
+                for error_param in params:
+                    stac_param = EODAGSearch.to_stac(error_param)
+                    exception.message = exception.message.replace(
+                        error_param, stac_param
+                    )
+                error_dict["message"] = exception.message
+
+            error_list.append(error_dict)
```

Comment:
Argument 1 to "append" of "list" has incompatible type "dict[str, str]"; expected "dict[str, str | int]"mypy[arg-type](https://mypy.readthedocs.io/en/stable/_refs.html#code-arg-type)